diff --git a/build/CheqdModule.js b/build/CheqdModule.js
index 1a0b97a9a8cc233f0fe94d0b80c1847ec7022865..47ac40340d5471acc988fe800f84c937d9b55a4b 100644
--- a/build/CheqdModule.js
+++ b/build/CheqdModule.js
@@ -26,7 +26,7 @@ class CheqdModule {
     async initialize(agentContext) {
         // not required
         const cheqdLedgerService = agentContext.dependencyManager.resolve(ledger_1.CheqdLedgerService);
-        await cheqdLedgerService.connect();
+        void cheqdLedgerService.connect();
     }
 }
 exports.CheqdModule = CheqdModule;
