diff --git a/build/shared/callbacks.js b/build/shared/callbacks.js
index 861794bc79a02c26a73d0f4403f0c8113aa395d7..3d179571ed346fba8bb371ccefef96e8ee4ecee5 100644
--- a/build/shared/callbacks.js
+++ b/build/shared/callbacks.js
@@ -201,8 +201,8 @@ function getOid4vcEncryptJweCallback(agentContext) {
             recipientKey: key,
             header: { kid: jweEncryptor.publicJwk.kid },
             encryptionAlgorithm: jweEncryptor.enc,
-            apu: jweEncryptor.apu ? core_2.TypedArrayEncoder.toBase64URL(core_2.TypedArrayEncoder.fromString(jweEncryptor.apu)) : undefined,
-            apv: jweEncryptor.apv ? core_2.TypedArrayEncoder.toBase64URL(core_2.TypedArrayEncoder.fromString(jweEncryptor.apv)) : undefined,
+            apu: jweEncryptor.apu,
+            apv: jweEncryptor.apv,
         });
         return { encryptionJwk: jweEncryptor.publicJwk, jwe };
     };
