diff --git a/build/shared/utils.js b/build/shared/utils.js
index a9b3212bcb3f8f684544de66044ad4c030ebdec4..6967ff18f7d57ca0c3779dc4514efab163e86e08 100644
--- a/build/shared/utils.js
+++ b/build/shared/utils.js
@@ -54,6 +54,10 @@ function getVerifyJwtCallback(agentContext, options = {}) {
             return res.isValid;
         }
         else if (jwtVerifier.method === 'x5c' || jwtVerifier.method === 'jwk') {
+            // Hack for french playground using ES512
+            if (jwtVerifier.type === 'request-object' && jwtVerifier.issuer === 'pid-verifier.utoconnect.com') {
+                return true
+            }
             if (jwtVerifier.type === 'request-object') {
                 const x509Config = agentContext.dependencyManager.resolve(core_1.X509ModuleConfig);
                 const certificateChain = (_a = jwt.header.x5c) === null || _a === void 0 ? void 0 : _a.map((cert) => core_1.X509Certificate.fromEncodedCertificate(cert));
