diff --git a/dist/index.js b/dist/index.js
index 704187eec354784a9d14e767555a559133b877f8..d53019eceb29d7d86cb4bb0f628aad19ee7e7efa 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -3667,11 +3667,8 @@ var limitDisclosureToDeviceRequestNameSpaces = (mdoc, deviceRequestNameSpaces) =
     const nsAttrs = mdoc.issuerSigned.nameSpaces.get(nameSpace) ?? [];
     const digests = Array.from(nameSpaceFields.entries()).map(([elementIdentifier, _]) => {
       const digest = prepareDigest(elementIdentifier, nsAttrs);
-      if (!digest) {
-        throw new Error(`No matching field found for '${elementIdentifier}'`);
-      }
       return digest;
-    });
+    }).filter((d) => d !== null);
     nameSpaces.set(nameSpace, digests);
   }
   return nameSpaces;
