diff --git a/build/modules/kms/KeyManagementApi.js b/build/modules/kms/KeyManagementApi.js
index bf5e3a50f6cfd4df3fd328d5006f1a5384dac1c9..4b84a07786d7786523f726f68d9f4a22dfacad55 100644
--- a/build/modules/kms/KeyManagementApi.js
+++ b/build/modules/kms/KeyManagementApi.js
@@ -212,9 +212,9 @@ let KeyManagementApi = class KeyManagementApi {
                 };
         }
         if (operation) {
-            throw new KeyManagementError_1.KeyManagementError(`No key management service supports ${(0, options_1.getKmsOperationHumanDescription)(operation)} that has a key with keyId '${keyId}'`);
+            throw new KeyManagementKeyNotFoundError_1.KeyManagementKeyNotFoundError(keyId, this.keyManagementConfig.backends.map(b => b.backend));
         }
-        throw new KeyManagementError_1.KeyManagementError(`No key management service has a key with keyId '${keyId}'`);
+        throw new KeyManagementKeyNotFoundError_1.KeyManagementKeyNotFoundError(keyId, this.keyManagementConfig.backends.map(b => b.backend));
     }
     /**
      * Get the kms backend for a specific operation.
diff --git a/build/modules/sd-jwt-vc/SdJwtVcService.js b/build/modules/sd-jwt-vc/SdJwtVcService.js
index 1e20be41bee98117d57ee01706ca9c851f862c9f..365112ea1105595aa1fd7c582fe6dbab841d0756 100644
--- a/build/modules/sd-jwt-vc/SdJwtVcService.js
+++ b/build/modules/sd-jwt-vc/SdJwtVcService.js
@@ -101,7 +101,8 @@ let SdJwtVcService = class SdJwtVcService {
             value: d.value,
         })), presentationFrame);
         const [jwt] = compactSdJwtVc.split('~');
-        const sdJwt = `${jwt}~${requiredDisclosures.map((d) => d.encoded).join('~')}~`;
+        const disclosuresString = requiredDisclosures.length > 0 ? `${requiredDisclosures.map((d) => d.encoded).join('~')}~` : ''
+        const sdJwt = `${jwt}~${disclosuresString}`
         const disclosedDecoded = (0, decodeSdJwtVc_1.decodeSdJwtVc)(sdJwt);
         return disclosedDecoded;
     }
