diff --git a/node_modules/@aries-framework/openid4vc-client/build/OpenId4VcClientService.js b/node_modules/@aries-framework/openid4vc-client/build/OpenId4VcClientService.js
index dcfde21..4fe5dcf 100644
--- a/node_modules/@aries-framework/openid4vc-client/build/OpenId4VcClientService.js
+++ b/node_modules/@aries-framework/openid4vc-client/build/OpenId4VcClientService.js
@@ -195,14 +195,15 @@ let OpenId4VcClientService = class OpenId4VcClientService {
             throw new core_1.AriesFrameworkError('Did not receive a successful credential response');
         }
         const credential = core_1.JsonTransformer.fromJSON(credentialResponse.successBody.credential, core_1.W3cVerifiableCredential);
+        console.log("Not verifying credential due to https://github.com/hyperledger/aries-framework-javascript/discussions/1432")
         // verify the signature
-        const result = await this.w3cCredentialService.verifyCredential(agentContext, {
-            credential,
-            verifyRevocationState: options.verifyRevocationState,
-        });
-        if (result && !result.verified) {
-            throw new core_1.AriesFrameworkError(`Failed to validate credential, error = ${result.error}`);
-        }
+        // const result = await this.w3cCredentialService.verifyCredential(agentContext, {
+        //     credential,
+        //     verifyRevocationState: options.verifyRevocationState,
+        // });
+        // if (result && !result.verified) {
+        //     throw new core_1.AriesFrameworkError(`Failed to validate credential, error = ${result.error}`);
+        // }
         const storedCredential = await this.w3cCredentialService.storeCredential(agentContext, {
             credential,
         });
