diff --git a/build/cjs/modules/did.js b/build/cjs/modules/did.js
index ecf1d42681c75765f95bee42b674a5206469033b..1fc8db7f1303df7341755113c38efb17f6d1d99f 100644
--- a/build/cjs/modules/did.js
+++ b/build/cjs/modules/did.js
@@ -12,7 +12,7 @@ exports.MsgDeactiveDidDocResponseEncodeObject = MsgDeactiveDidDocResponseEncodeO
 const stargate_cjs_1 = require("@cosmjs/stargate-cjs");
 const _1 = require("./_");
 const types_1 = require("../types");
-const index_1 = require("@cheqd/ts-proto-cjs/cheqd/did/v2/index");
+const index_1 = require("@cheqd/ts-proto/cheqd/did/v2/index");
 const uuid_cjs_1 = require("uuid-cjs");
 const utils_cjs_1 = require("@cosmjs/utils-cjs");
 exports.defaultDidExtensionKey = 'did';
diff --git a/build/cjs/modules/feeabstraction.js b/build/cjs/modules/feeabstraction.js
index 6c6c8248da1b66fe8a0b5f97ee093d4cf3322558..43fde98524168cc88dd03168edf4a23cc81ac5ed 100644
--- a/build/cjs/modules/feeabstraction.js
+++ b/build/cjs/modules/feeabstraction.js
@@ -15,7 +15,7 @@ exports.isMsgUpdateHostZoneEncodeObject = isMsgUpdateHostZoneEncodeObject;
 exports.isMsgUpdateHostZoneResponseEncodeObject = isMsgUpdateHostZoneResponseEncodeObject;
 exports.isMsgUpdateParamsEncodeObject = isMsgUpdateParamsEncodeObject;
 exports.isMsgUpdateParamsResponseEncodeObject = isMsgUpdateParamsResponseEncodeObject;
-const index_1 = require("@cheqd/ts-proto-cjs/feeabstraction/feeabs/v1beta1/index");
+const index_1 = require("@cheqd/ts-proto/feeabstraction/feeabs/v1beta1/index");
 const _1 = require("./_");
 const stargate_cjs_1 = require("@cosmjs/stargate-cjs");
 const tx_1 = require("cosmjs-types/cosmos/gov/v1/tx");
diff --git a/build/cjs/modules/feemarket.js b/build/cjs/modules/feemarket.js
index 5f70961770bc7ea21a12a0754ca890a640c6161e..e1c94f88943211aa4a35980ac95fa849cddd1395 100644
--- a/build/cjs/modules/feemarket.js
+++ b/build/cjs/modules/feemarket.js
@@ -4,7 +4,7 @@ exports.FeemarketModule = exports.setupFeemarketExtension = exports.defaultGasPr
 exports.isGasPriceEncodeObject = isGasPriceEncodeObject;
 exports.isGasPricesEncodeObject = isGasPricesEncodeObject;
 exports.isParamsEncodeObject = isParamsEncodeObject;
-const index_js_1 = require("@cheqd/ts-proto-cjs/feemarket/feemarket/v1/index.js");
+const index_js_1 = require("@cheqd/ts-proto/feemarket/feemarket/v1/index.js");
 const stargate_cjs_1 = require("@cosmjs/stargate-cjs");
 const _1 = require("./_");
 const utils_1 = require("../utils");
diff --git a/build/cjs/modules/resource.js b/build/cjs/modules/resource.js
index 83567497eb21a2768cc1f322b3c1e44f72f0fd3c..57798df1bf47e251c1b0192cd45af82dda7e171f 100644
--- a/build/cjs/modules/resource.js
+++ b/build/cjs/modules/resource.js
@@ -5,7 +5,7 @@ exports.isMsgCreateResourceEncodeObject = isMsgCreateResourceEncodeObject;
 const _1 = require("./_");
 const signer_1 = require("../signer");
 const types_1 = require("../types");
-const v2_1 = require("@cheqd/ts-proto-cjs/cheqd/resource/v2");
+const v2_1 = require("@cheqd/ts-proto/cheqd/resource/v2");
 const stargate_cjs_1 = require("@cosmjs/stargate-cjs");
 const uint8arrays_cjs_1 = require("uint8arrays-cjs");
 const browser_1 = require("file-type-cjs/browser");
diff --git a/build/cjs/signer.js b/build/cjs/signer.js
index 70952cf919696820ec0942d7813f224228621ad4..349d87656680649705adb278d7f8e493ca7279ca 100644
--- a/build/cjs/signer.js
+++ b/build/cjs/signer.js
@@ -11,7 +11,7 @@ const proto_signing_cjs_1 = require("@cosmjs/proto-signing-cjs");
 const stargate_cjs_1 = require("@cosmjs/stargate-cjs");
 const tendermint_rpc_cjs_1 = require("@cosmjs/tendermint-rpc-cjs");
 const registry_1 = require("./registry");
-const v2_1 = require("@cheqd/ts-proto-cjs/cheqd/did/v2");
+const v2_1 = require("@cheqd/ts-proto/cheqd/did/v2");
 const types_1 = require("./types");
 const did_jwt_cjs_1 = require("did-jwt-cjs");
 const utils_cjs_1 = require("@cosmjs/utils-cjs");
diff --git a/build/cjs/utils.js b/build/cjs/utils.js
index 130d4cd4efa9e6490ba5b47d5bee5e992d482146..f4b7065fefda3010b9793bb34bd08bfe2b5d4f59 100644
--- a/build/cjs/utils.js
+++ b/build/cjs/utils.js
@@ -32,9 +32,9 @@ const crypto_cjs_1 = require("@cosmjs/crypto-cjs");
 const amino_cjs_1 = require("@cosmjs/amino-cjs");
 const secp256k1_cjs_1 = __importDefault(require("secp256k1-cjs"));
 const uuid_cjs_1 = require("uuid-cjs");
-const v2_1 = require("@cheqd/ts-proto-cjs/cheqd/did/v2");
+const v2_1 = require("@cheqd/ts-proto/cheqd/did/v2");
 const did_1 = require("./modules/did");
-const v2_2 = require("@cheqd/ts-proto-cjs/cheqd/resource/v2");
+const v2_2 = require("@cheqd/ts-proto/cheqd/resource/v2");
 const encoding_cjs_1 = require("@cosmjs/encoding-cjs");
 const stargate_cjs_1 = require("@cosmjs/stargate-cjs");
 const exponential_backoff_cjs_1 = require("exponential-backoff-cjs");
