diff --git a/build/wallet/AskarBaseWallet.js b/build/wallet/AskarBaseWallet.js
index f4352d639a511347d9a087f4ba5c5afddec477ce..0f5fa56a6099ffe1fdf5e5f604626de309e58e62 100644
--- a/build/wallet/AskarBaseWallet.js
+++ b/build/wallet/AskarBaseWallet.js
@@ -115,8 +115,8 @@ class AskarBaseWallet {
                 const secureEnvironment = (0, secureEnvironment_1.importSecureEnvironment)();
                 const kid = core_1.utils.uuid();
                 // Generate a hardware-backed P-256 keypair
-                secureEnvironment.generateKeypair(kid);
-                const publicKeyBytes = secureEnvironment.getPublicBytesForKeyId(kid);
+                await secureEnvironment.generateKeypair(kid);
+                const publicKeyBytes = await secureEnvironment.getPublicBytesForKeyId(kid);
                 const publicKeyBase58 = core_1.TypedArrayEncoder.toBase58(publicKeyBytes);
                 await this.storeSecureEnvironmentKeyById({
                     keyType,
