diff --git a/build/openid4vc-holder/OpenId4VcHolderApi.js b/build/openid4vc-holder/OpenId4VcHolderApi.js
index 9605e9a095d780e666fd6afbd0b95c41b44fbd69..070a203d2a56f5439cd278daa7bec54f69382147 100644
--- a/build/openid4vc-holder/OpenId4VcHolderApi.js
+++ b/build/openid4vc-holder/OpenId4VcHolderApi.js
@@ -140,7 +140,7 @@ let OpenId4VcHolderApi = class OpenId4VcHolderApi {
      * @param options.tokenResponse Obtained through @see requestAccessToken
      */
     async requestCredentials(options) {
-        const { resolvedCredentialOffer, cNonce, accessToken, dpop, clientId } = options, credentialRequestOptions = __rest(options, ["resolvedCredentialOffer", "cNonce", "accessToken", "dpop", "clientId"]);
+        const { resolvedCredentialOffer, cNonce, accessToken, dpop, clientId, additionalProofOfPossessionPayloadClaims, additionalCredentialRequestPayloadClaims, customFormat, popCallback, getCreateJwtCallback,customBody,skipSdJwtVcValidation } = options, credentialRequestOptions = __rest(options, ["resolvedCredentialOffer", "cNonce", "accessToken", "dpop", "clientId"]);
         return this.openId4VciHolderService.acceptCredentialOffer(this.agentContext, {
             resolvedCredentialOffer,
             acceptCredentialOfferOptions: credentialRequestOptions,
@@ -148,6 +148,13 @@ let OpenId4VcHolderApi = class OpenId4VcHolderApi {
             cNonce,
             dpop,
             clientId,
+            additionalCredentialRequestPayloadClaims,
+            additionalProofOfPossessionPayloadClaims,
+            customFormat,
+            popCallback,
+            getCreateJwtCallback,
+            customBody,
+            skipSdJwtVcValidation
         });
     }
     /**
diff --git a/build/openid4vc-holder/OpenId4VciHolderService.js b/build/openid4vc-holder/OpenId4VciHolderService.js
index 83d55fd85edd379c9cd01947304478934c16212b..ccf9ebc0272b9c0642602658d496463b69fd72a9 100644
--- a/build/openid4vc-holder/OpenId4VciHolderService.js
+++ b/build/openid4vc-holder/OpenId4VciHolderService.js
@@ -129,7 +129,7 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
         const authDetails = offeredCredentials
             .map((credential) => this.getAuthDetailsFromOfferedCredential(credential, authDetailsLocation))
             .filter((authDetail) => authDetail !== undefined);
-        const { clientId, redirectUri, scope } = authCodeFlowOptions;
+        const { clientId, redirectUri, scope, customHeaders } = authCodeFlowOptions;
         const vciClientState = {
             state: {
                 credentialOffer: resolvedCredentialOffer.credentialOfferRequestWithBaseUrl,
@@ -154,6 +154,7 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
                 authorizationDetails: authDetails,
                 parMode: oid4vci_common_1.PARMode.AUTO,
             },
+            headers: customHeaders
         });
         return Object.assign(Object.assign({}, authCodeFlowOptions), { codeVerifier,
             authorizationRequestUri });
@@ -176,8 +177,12 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
             return undefined;
         const alg = dpopSigningAlgValuesSupported.find((alg) => (0, core_1.getJwkClassFromJwaSignatureAlgorithm)(alg));
         let jwk;
+        let getCreateJwtCallback = utils_1.getCreateJwtCallback;
         if (resourceRequestOptions) {
             jwk = resourceRequestOptions.jwk;
+            if(resourceRequestOptions.getCreateJwtCallback !== null && resourceRequestOptions.getCreateJwtCallback !== void 0) {
+              getCreateJwtCallback = resourceRequestOptions.getCreateJwtCallback;
+            }
         }
         else {
             const JwkClass = alg ? (0, core_1.getJwkClassFromJwaSignatureAlgorithm)(alg) : undefined;
@@ -191,20 +196,20 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
             jwtIssuer: { alg: alg, jwk: jwk.toJson() },
             dPoPSigningAlgValuesSupported: dpopSigningAlgValuesSupported,
             jwtPayloadProps: (_c = resourceRequestOptions === null || resourceRequestOptions === void 0 ? void 0 : resourceRequestOptions.jwtPayloadProps) !== null && _c !== void 0 ? _c : {},
-            createJwtCallback: (0, utils_1.getCreateJwtCallback)(agentContext),
+            createJwtCallback: getCreateJwtCallback(agentContext),
         };
         return createDPoPOpts;
     }
     async requestAccessToken(agentContext, options) {
         var _a, _b, _c, _d;
-        const { resolvedCredentialOffer, txCode, resolvedAuthorizationRequest, code } = options;
+        const { resolvedCredentialOffer, txCode, resolvedAuthorizationRequest, code, customBody, dPopKeyJwk, getCreateJwtCallback } = options;
         const { metadata, credentialOfferRequestWithBaseUrl } = resolvedCredentialOffer;
         // acquire the access token
         let accessTokenResponse;
         const accessTokenClient = new oid4vci_client_1.AccessTokenClient();
-        const createDPoPOpts = await this.getCreateDpopOptions(agentContext, metadata);
-        let dpopJwk;
-        if (createDPoPOpts) {
+        const createDPoPOpts = await this.getCreateDpopOptions(agentContext, metadata, dPopKeyJwk === null || dPopKeyJwk === void 0 ? undefined : { jwk: dPopKeyJwk, getCreateJwtCallback });
+        let dpopJwk = dPopKeyJwk;
+        if ((dpopJwk == null || dpopJwk === void 0) && createDPoPOpts) {
             if (!createDPoPOpts.jwtIssuer.jwk.kty) {
                 throw new core_1.CredoError('Missing required key type (kty) in the jwk.');
             }
@@ -228,6 +233,7 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
                 credentialOffer: { credential_offer: credentialOfferRequestWithBaseUrl.credential_offer },
                 pin: txCode,
                 createDPoPOpts,
+                customBody
             });
         }
         if (!accessTokenResponse.successBody) {
@@ -238,7 +244,8 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
     }
     async acceptCredentialOffer(agentContext, options) {
         var _a, _b, _c, _d, _e, _f;
-        const { resolvedCredentialOffer, acceptCredentialOfferOptions } = options;
+        const { resolvedCredentialOffer, acceptCredentialOfferOptions, customFormat, additionalProofOfPossessionPayloadClaims, additionalCredentialRequestPayloadClaims, getCreateJwtCallback, popCallback, customBody, skipSdJwtVcValidation } = options;
+
         const { metadata, version, offeredCredentialConfigurations } = resolvedCredentialOffer;
         const { credentialsToRequest, credentialBindingResolver, verifyCredentialStatus } = acceptCredentialOfferOptions;
         if ((credentialsToRequest === null || credentialsToRequest === void 0 ? void 0 : credentialsToRequest.length) === 0) {
@@ -293,11 +300,17 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
             // Create the proof of possession
             const proofOfPossessionBuilder = oid4vci_client_1.ProofOfPossessionBuilder.fromAccessTokenResponse({
                 accessTokenResponse: tokenResponse,
-                callbacks: { signCallback: this.proofOfPossessionSignCallback(agentContext) },
+                callbacks: { signCallback: popCallback === null || popCallback === void 0 ? this.proofOfPossessionSignCallback(agentContext) : popCallback },
                 version,
             })
                 .withEndpointMetadata(metadata)
                 .withAlg(signatureAlgorithm);
+            if(additionalProofOfPossessionPayloadClaims !== null && additionalProofOfPossessionPayloadClaims !== void 0) {
+              if(additionalProofOfPossessionPayloadClaims.iss !== null) {
+                proofOfPossessionBuilder.withIssuer(additionalProofOfPossessionPayloadClaims.iss)
+              }
+              // proofOfPossessionBuilder.withJwt({header: {}, payload: additionalProofOfPossessionPayloadClaims })
+            }
             // TODO: what if auth flow using did, and the did is different from client id. We now use the client_id
             if (credentialBinding.method === 'did') {
                 proofOfPossessionBuilder.withClientId((0, core_1.parseDid)(credentialBinding.didUrl).did).withKid(credentialBinding.didUrl);
@@ -323,11 +336,13 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
                 .withVersion(version)
                 .withCredentialEndpoint(metadata.credential_endpoint)
                 .withToken(tokenResponse.access_token);
+            if(customFormat) { credentialRequestBuilder.withFormat(customFormat) }
             const credentialRequestClient = credentialRequestBuilder.build();
             const createDpopOpts = tokenResponse.dpop
                 ? await this.getCreateDpopOptions(agentContext, metadata, {
                     jwk: tokenResponse.dpop.jwk,
                     jwtPayloadProps: { accessToken: tokenResponse.access_token, nonce: (_e = tokenResponse.dpop) === null || _e === void 0 ? void 0 : _e.nonce },
+                    getCreateJwtCallback
                 })
                 : undefined;
             const credentialResponse = await credentialRequestClient.acquireCredentialsUsingProof({
@@ -335,13 +350,20 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
                 credentialTypes: (0, issuerMetadataUtils_1.getTypesFromCredentialSupported)(offeredCredentialConfiguration),
                 format: offeredCredentialConfiguration.format,
                 createDPoPOpts: createDpopOpts,
+                customBody: customBody,
+                additionalRequestClaims: additionalCredentialRequestPayloadClaims
             });
+
+            if(customBody !== null && customBody !== void 0) {
+              return [credentialResponse.successBody.credential]
+            }
             newCNonce = (_f = credentialResponse.successBody) === null || _f === void 0 ? void 0 : _f.c_nonce;
             // Create credential, but we don't store it yet (only after the user has accepted the credential)
             const credential = await this.handleCredentialResponse(agentContext, credentialResponse, {
                 verifyCredentialStatus: verifyCredentialStatus !== null && verifyCredentialStatus !== void 0 ? verifyCredentialStatus : false,
                 credentialIssuerMetadata: metadata.credentialIssuerMetadata,
-                format: offeredCredentialConfiguration.format,
+                format: customFormat !== null && customFormat !== void 0 ? customFormat : offeredCredentialConfiguration.format,
+                skipSdJwtVcValidation
             });
             this.logger.debug('Full credential', credential);
             receivedCredentials.push(credential);
@@ -487,14 +509,18 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
             if (typeof credentialResponse.successBody.credential !== 'string')
                 throw new core_1.CredoError(`Received a credential of format ${shared_1.OpenId4VciCredentialFormatProfile.SdJwtVc}, but the credential is not a string. ${JSON.stringify(credentialResponse.successBody.credential)}`);
             const sdJwtVcApi = agentContext.dependencyManager.resolve(core_1.SdJwtVcApi);
-            const verificationResult = await sdJwtVcApi.verify({
-                compactSdJwtVc: credentialResponse.successBody.credential,
-            });
-            if (!verificationResult.isValid) {
-                agentContext.config.logger.error('Failed to validate credential', { verificationResult });
-                throw new core_1.CredoError(`Failed to validate sd-jwt-vc credential. Results = ${JSON.stringify(verificationResult)}`);
+            if(options.skipSdJwtVcValidation === null || options.skipSdJwtVcValidation === void 0 || options.skipSdJwtVcValidation === false) { 
+              const verificationResult = await sdJwtVcApi.verify({
+                  compactSdJwtVc: credentialResponse.successBody.credential,
+              });
+              if (!verificationResult.isValid) {
+                  agentContext.config.logger.error('Failed to validate credential', { verificationResult });
+                  throw new core_1.CredoError(`Failed to validate sd-jwt-vc credential. Results = ${JSON.stringify(verificationResult)}`);
+              }
+              return { credential: verificationResult.sdJwtVc, notificationMetadata };
+            } else {
+            return { credential: sdJwtVcApi.fromCompact(credentialResponse.successBody.credential), notificationMetadata };
             }
-            return { credential: verificationResult.sdJwtVc, notificationMetadata };
         }
         else if (format === shared_1.OpenId4VciCredentialFormatProfile.JwtVcJson ||
             format === shared_1.OpenId4VciCredentialFormatProfile.JwtVcJsonLd) {
@@ -526,6 +552,7 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
                 throw new core_1.CredoError('Received Mdoc credential in an unsupported format.');
             }
             const mdoc = core_1.Mdoc.fromIssuerSignedBase64(credentialResponse.successBody.credential);
+            console.log('running mdoc verification now')
             const result = await this.mdocService.verify(agentContext, { mdoc });
             if (!result.isValid) {
                 agentContext.config.logger.error('Failed to validate credential', { result });
