diff --git a/build/shared/issuerMetadataUtils.js b/build/shared/issuerMetadataUtils.js
index 975341a9efacacd2859f4a992e643f9b43132fb6..0ac6a2aafd743534645cf88e8b39f3ff9e91fb9b 100644
--- a/build/shared/issuerMetadataUtils.js
+++ b/build/shared/issuerMetadataUtils.js
@@ -56,7 +56,8 @@ function credentialConfigurationSupportedToCredentialSupported(id, config) {
         }
         return Object.assign(Object.assign({}, baseConfig), { format: config.format, vct: config.vct, claims: config.claims });
     }
-    throw new core_1.CredoError(`Unsupported credential format ${config.format}`);
+    return baseConfig
+    // throw new core_1.CredoError(`Unsupported credential format ${config.format}`);
 }
 function credentialSupportedToCredentialConfigurationSupported(agentContext, credentialSupported) {
     var _a, _b;
@@ -94,7 +95,8 @@ function credentialSupportedToCredentialConfigurationSupported(agentContext, cre
     else if (credentialSupported.format === 'vc+sd-jwt') {
         return Object.assign(Object.assign({}, baseCredentialConfigurationSupported), { format: credentialSupported.format, vct: credentialSupported.vct, claims: credentialSupported.claims });
     }
-    throw new core_1.CredoError(`Unsupported credential format ${credentialSupported.format}`);
+    return baseCredentialConfigurationSupported
+    // throw new core_1.CredoError(`Unsupported credential format ${credentialSupported.format}`);
 }
 function credentialsSupportedV13ToV11(credentialConfigurationSupported) {
     const credentialsSupportedWithId = [];
