diff --git a/build/shared/issuerMetadataUtils.js b/build/shared/issuerMetadataUtils.js
index fb5a81e3beb103d7e411a3d687290445a7bb2d20..eb14ae9ff380aaae2016f411a1eacec0bfdb8138 100644
--- a/build/shared/issuerMetadataUtils.js
+++ b/build/shared/issuerMetadataUtils.js
@@ -57,7 +57,10 @@ function credentialConfigurationSupportedToCredentialSupported(id, config) {
         }
         return Object.assign(Object.assign({}, baseConfig), { format: config.format, vct: config.vct, claims: config.claims });
     }
-    throw new core_1.CredoError(`Unsupported credential format ${config.format}`);
+    return {
+        ...baseConfig,
+        ...config
+    }
 }
 function credentialSupportedToCredentialConfigurationSupported(agentContext, credentialSupported) {
     var _a, _b;
