diff --git a/ReactNativeFs.podspec b/ReactNativeFs.podspec
index f9deeda044ff83da5291d3eb32539aa321f145a0..d08e887d75ef25875c4b970ca17afe305d0a48f1 100644
--- a/ReactNativeFs.podspec
+++ b/ReactNativeFs.podspec
@@ -14,7 +14,8 @@ Pod::Spec.new do |s|
   s.source       = { :git => "https://github.com/birdofpreyru/react-native-fs.git", :tag => "#{s.version}" }
 
   s.resource_bundles = { 'RNFS_PrivacyInfo' => 'ios/PrivacyInfo.xcprivacy' }
-  s.frameworks = "AVFoundation", "Photos"
+  # NOTE: see https://github.com/birdofpreyru/react-native-fs/issues/130
+  s.frameworks = "AVFoundation", "Photos", "UniformTypeIdentifiers"
   s.source_files = "ios/**/*.{h,m,mm,swift,cpp}"
 
   # Necessary for background downloads to work,
