diff --git a/android/build.gradle b/android/build.gradle
index 9017d77eb32573415c7922cd83dae5cd792a885f..759b67025c2414b86ee204e1c5a52562da7b6caf 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -59,6 +59,10 @@ def getExt(name) {
   return rootProject.ext.get(name)
 }
 
+def getExtWithFallback(prop, fallback) {
+  return rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
 def resolveBuildType() {
     def buildType = "debug"
     tasks.all({ task ->
@@ -81,7 +85,7 @@ android {
   }
   
   defaultConfig {
-    minSdkVersion 21 
+    minSdkVersion getExtWithFallback('minSdkVersion', 21)
     targetSdkVersion getExt('targetSdkVersion')
     
     externalNativeBuild {
