diff --git a/android/build.gradle b/android/build.gradle
index 2cc06db34a3043bea273bd6f9a69b1151711a433..aaa4a96e4a42d2e5f5880d58a2198c6fb9c0808d 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -59,6 +59,10 @@ def getExt(name) {
   return rootProject.ext.get(name)
 }
 
+def getExtWithFallback(prop, fallback) {
+  return rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
 def resolveBuildType() {
     def buildType = "debug"
     tasks.all({ task ->
@@ -81,7 +85,7 @@ android {
   }
   
   defaultConfig {
-    minSdkVersion 21 
+    minSdkVersion getExtWithFallback('minSdkVersion', 21)
     targetSdkVersion getExt('targetSdkVersion')
     
     externalNativeBuild {
