diff --git a/aries-askar.podspec b/aries-askar.podspec
index 655864cae6906bc283e09abde94b1496b0311adf..babed5816ed65975e9a568d6444afb825059414a 100644
--- a/aries-askar.podspec
+++ b/aries-askar.podspec
@@ -16,7 +16,8 @@ Pod::Spec.new do |s|
   s.header_mappings_dir = "cpp"
 
   s.pod_target_xcconfig = {
-    :USE_HEADERMAP => "No"
+    :USE_HEADERMAP => "No",
+    'CLANG_CXX_LANGUAGE_STANDARD' => "c++20",
   }
 
   s.ios.vendored_frameworks = "native/mobile/ios/aries_askar.xcframework"
