diff --git a/android/build.gradle b/android/build.gradle
index 9017d77eb32573415c7922cd83dae5cd792a885f..759b67025c2414b86ee204e1c5a52562da7b6caf 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -59,6 +59,10 @@ def getExt(name) {
   return rootProject.ext.get(name)
 }
 
+def getExtWithFallback(prop, fallback) {
+  return rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
 def resolveBuildType() {
     def buildType = "debug"
     tasks.all({ task ->
@@ -81,7 +85,7 @@ android {
   }
   
   defaultConfig {
-    minSdkVersion 21 
+    minSdkVersion getExtWithFallback('minSdkVersion', 21)
     targetSdkVersion getExt('targetSdkVersion')
     
     externalNativeBuild {
diff --git a/indy-vdr.podspec b/indy-vdr.podspec
index 8488810b804d6516f6a0d5a62020516e579ba884..78219c1ab3de60cf0382a7f479e23090895f0ee6 100644
--- a/indy-vdr.podspec
+++ b/indy-vdr.podspec
@@ -16,7 +16,8 @@ Pod::Spec.new do |s|
   s.header_mappings_dir = "cpp"
 
   s.pod_target_xcconfig = {
-    :USE_HEADERMAP => "No"
+    :USE_HEADERMAP => "No",
+    'CLANG_CXX_LANGUAGE_STANDARD' => "c++20",
   }
 
   s.ios.vendored_frameworks = "native/mobile/ios/indy_vdr.xcframework"
