diff --git a/dist/index.js b/dist/index.js
index cf4a320d257f572d8063201f0d47cad68887e158..6463688dd29ffbe469daa6f0dbd3123ed381bb08 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1060,7 +1060,7 @@ function extractJwksFromClientMetadata(clientMetadata) {
   const parsed = zJarmClientMetadataParsed.parse(clientMetadata);
   const encryptionAlg = parsed.client_metadata.authorization_encrypted_response_enc;
   const signingAlg = parsed.client_metadata.authorization_signed_response_alg;
-  const encJwk = clientMetadata.jwks.keys.find((key) => key.use === "enc" && key.alg === encryptionAlg) ?? clientMetadata.jwks.keys.find((key) => key.use === "enc");
+  const encJwk = clientMetadata.jwks.keys.find((key) => key.use === "enc" && key.alg === encryptionAlg) ?? clientMetadata.jwks.keys.find((key) => key.use === "enc") ?? clientMetadata.jwks.keys?.[0]
   const sigJwk = clientMetadata.jwks.keys.find((key) => key.use === "sig" && key.alg === signingAlg) ?? clientMetadata.jwks.keys.find((key) => key.use === "sig");
   return { encJwk, sigJwk };
 }
diff --git a/dist/index.mjs b/dist/index.mjs
index 91c93350e6087f4a267fb006857433bb98b7e7b9..61c0b7a4fdff93211e2051c5606a743db2294938 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -1027,7 +1027,7 @@ function extractJwksFromClientMetadata(clientMetadata) {
   const parsed = zJarmClientMetadataParsed.parse(clientMetadata);
   const encryptionAlg = parsed.client_metadata.authorization_encrypted_response_enc;
   const signingAlg = parsed.client_metadata.authorization_signed_response_alg;
-  const encJwk = clientMetadata.jwks.keys.find((key) => key.use === "enc" && key.alg === encryptionAlg) ?? clientMetadata.jwks.keys.find((key) => key.use === "enc");
+  const encJwk = clientMetadata.jwks.keys.find((key) => key.use === "enc" && key.alg === encryptionAlg) ?? clientMetadata.jwks.keys.find((key) => key.use === "enc") ?? clientMetadata.jwks.keys?.[0];
   const sigJwk = clientMetadata.jwks.keys.find((key) => key.use === "sig" && key.alg === signingAlg) ?? clientMetadata.jwks.keys.find((key) => key.use === "sig");
   return { encJwk, sigJwk };
 }
