diff --git a/dist/src/mdoc/model/device-response.js b/dist/src/mdoc/model/device-response.js
index 690940a79a8ea60bea82aac530825374577d309e..a74e93d6f8b49338e42b0e249978f3ece2e42b05 100644
--- a/dist/src/mdoc/model/device-response.js
+++ b/dist/src/mdoc/model/device-response.js
@@ -264,6 +264,7 @@ class DeviceResponse {
             jwk: this.devicePrivateKey,
         });
         sign1.signature = signature;
+        sign1.payload = null
         return { deviceSignature: sign1 };
     }
 }
diff --git a/dist/src/mdoc/model/device-signed-document.js b/dist/src/mdoc/model/device-signed-document.js
index b96a6bf841292c046047ccdd9b256a5cbf8a8562..852913f10ee0c60ce7577d488e1e8a45beb38455 100644
--- a/dist/src/mdoc/model/device-signed-document.js
+++ b/dist/src/mdoc/model/device-signed-document.js
@@ -19,24 +19,23 @@ class DeviceSignedDocument extends issuer_signed_document_js_1.IssuerSignedDocum
         const doc = super.prepare();
         const deviceSignature = this.deviceSigned.deviceAuth.deviceSignature?.getContentForEncoding();
         const deviceMac = this.deviceSigned.deviceAuth.deviceMac?.getContentForEncoding();
-        // detach payload
-        if (deviceMac) {
-            deviceMac[2] = undefined;
-        }
-        if (deviceSignature) {
-            deviceSignature[2] = undefined;
-        }
-        //
         doc.set('deviceSigned', {
             ...this.deviceSigned,
             nameSpaces: data_item_js_1.DataItem.fromData(this.deviceSigned.nameSpaces),
             // TODO: ERRORS MISSING
             deviceAuth: {
                 ...this.deviceSigned.deviceAuth,
-                deviceSignature,
+                deviceSignature: [...deviceSignature],
                 deviceMac,
             },
         });
+        // detach payload
+        if (deviceMac) {
+            deviceMac[2] = undefined;
+        }
+        if (deviceSignature) {
+            deviceSignature[2] = undefined;
+        }
         return doc;
     }
     /**
diff --git a/dist/src/mdoc/model/pex-limit-disclosure.js b/dist/src/mdoc/model/pex-limit-disclosure.js
index 2be88974a9e98b6699d9e8e76259f6d4f123f8dd..acce89c4bb6cf3bcad2e880d136645d4df51408c 100644
--- a/dist/src/mdoc/model/pex-limit-disclosure.js
+++ b/dist/src/mdoc/model/pex-limit-disclosure.js
@@ -6,7 +6,6 @@ const limitDisclosureToDeviceRequestNameSpaces = (mdoc, deviceRequestNameSpaces)
     for (const [nameSpace, nameSpaceFields] of Object.entries(deviceRequestNameSpaces)) {
         const nsAttrs = mdoc.issuerSigned.nameSpaces[nameSpace] ?? [];
         const digests = Object.entries(nameSpaceFields)
-            .filter(([_, disclose]) => disclose)
             .map(([elementIdentifier, _]) => {
             const digest = prepareDigest(elementIdentifier, nsAttrs);
             if (!digest) {
